server/src/modules/cats/cats.service.ts

```diff
@@ -1,45 +1,35 @@
-import { Injectable, Inject, NotFoundException } from '@nestjs/common';
+import { Injectable, NotFoundException } from '@nestjs/common';
+import { EntityRepository } from '@mikro-orm/postgresql';
+import { InjectRepository } from '@mikro-orm/nestjs';
 import { Cat } from 'src/models/cat/cat.model';
-import { Mouse } from 'src/models/mouse/mouse.model';
 import { CreateCatDto } from 'src/types/dto/cat.dto';
-import { Op } from 'sequelize';
+import { Mouse } from 'src/models/mouse/mouse.model';
 
 @Injectable()
 export class CatsService {
-  constructor(
-    @Inject('CATS_REPOSITORY')
-    private catsRepository: typeof Cat
-  ) { }
-
-  async create(createCatDto: CreateCatDto): Promise<Cat> {
+  constructor(@InjectRepository(Cat) private catRepository: EntityRepository<Cat>) { }
 
-    return this.catsRepository.create(createCatDto);
-  }
-
-  async findAll(filter: string): Promise<Cat[]> {
-    const whereCondition = filter
-      ? {
-        [Op.or]: [
-          { firstName: { [Op.iLike]: `%${filter}%` } },
-          { lastName: { [Op.iLike]: `%${filter}%` } },
-          { '$mice.name$': { [Op.iLike]: `%${filter}%` } },
-        ],
-      }
-      : {};
+  async findAll(filter?: string): Promise<Cat[]> {
+    if (filter) {
+      return await this.catRepository.find({
```

Comment:
return await Promise === return Promise

```suggestion
      return this.catRepository.find({
```